lightblue/src/main/java/org/esbtools/eventhandler/lightblue/LightblueDocumentEventRepository.java

```diff
@@ -92,30 +101,77 @@ public void addNewDocumentEvents(Collection<? extends DocumentEvent> documentEve
 
     @Override
     public List<LightblueDocumentEvent> retrievePriorityDocumentEventsUpTo(int maxEvents)
-            throws LightblueException {
-        try {
-            blockUntilLockAcquired(Locks.forDocumentEventsForEntities(entities));
+            throws Exception {
+        String[] typesToProcess = getSupportedAndEnabledEventTypes();
+        Integer documentEventsBatchSize = config.getDocumentEventsBatchSize();
+
+        if (typesToProcess.length == 0 || documentEventsBatchSize == null ||
+                documentEventsBatchSize == 0) {
+            logger.info("Not retrieving any document events because either there are no enabled " +
+                    "or supported types to process or documentEventBatchSize is 0. Supported " +
+                    "types are {}. Of those, enabled types are {}. " +
+                    "Document event batch size is {}.",
+                    supportedTypes, Arrays.toString(typesToProcess), documentEventsBatchSize);
+            return Collections.emptyList();
+        }
 
-            DocumentEventEntity[] documentEventEntities = lightblue
-                    .data(FindRequests.priorityDocumentEventsForEntitiesUpTo(entities, documentEventBatchSize))
-                    .parseProcessed(DocumentEventEntity[].class);
+        if (maxEvents == 0) {
+            return Collections.emptyList();
+        }
 
-            if (documentEventEntities.length == 0) {
-                return Collections.emptyList();
-            }
+        DocumentEventEntity[] documentEventEntities = lightblue
```

Comment:
You retrieve a batch of event documents here, and then try to optimize by matching identities. The batch may not contain all events of a particular identity because of the batch size restriction. A better approach would be is to get a batch of identities, lock them, and then load all the documents that were acquired. Then you'll have all documents of a particular identity at a given moment.